numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -4414,7 +4414,17 @@ PyArray_DescrFromType(int type)
 {
     PyArray_Descr *ret = NULL;
 
-    if (type < NPY_NTYPES) {
+    if (type < 0) {
```

Comment:
I tried a simple inline check in the if(type < NPY_NTYPES) branch, but then other branches tried to get in on the action, so..

Reply:
This seems like a good sanity check to add since `PyArray_DescrFromType(int type)` is an API function